### PR TITLE
feat(TypographyOverflow): improve calculation for displaying tooltip

### PR DESCRIPTION
**Pull request by bogdanalexe90**
[FX-1888]

### Description
POC for getting rid of `Ellipsis` component which triggers style recalculations on the first render.

ToDo:
- `interactive` doesn't work
- `tooltipDelay` doesn't work
-  add tests

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

[FX-1888]: https://toptal-core.atlassian.net/browse/FX-1888

### CODE REVIEW

File: packages/picasso-lab/src/TypographyOverflow/TypographyOverflow.tsx [hunk 1]
```diff
@@ -27,6 +26,15 @@ const useStyles = makeStyles<Theme, Props>(styles, {
...
 })
 
+/**
+ * Pixel value of font render space correction.
+ * It's individual for different fonts, so it won't work for 100% cases,
+ * but it allows us to be much closer to actual overflow detection while calculating.
+ * Tolerance of the render could be 0-2px depending on the font that is used,
+ * and also affected by the right-padding added at Ellipsis component.
+ */
+const FONT_RENDER_CORRECTION = 0.475
```
Comment (denieler) [inline packages/picasso-lab/src/TypographyOverflow/TypographyOverflow.tsx:36]: I want to cry when I see this 😃

Comment (OleksandrNechai) [reply packages/picasso-lab/src/TypographyOverflow/TypographyOverflow.tsx:36]: Yeah... At least this one comes from here https://github.com/toptal/picasso/blob/master/packages/picasso-lab/src/Ellipsis/use-ellipsis.ts#L19

File: packages/picasso-lab/src/Ellipsis/Ellipsis.tsx [hunk 2]
```diff
@@ -1,51 +0,0 @@
...
-       * Workaround for `text-overflow:ellipsis`.
-       * Browser can show ellipsis at the end, even if it's not required -
-       * it's possible if the text has the very same width as the container
-       * (ellipsis width will be the same as the width of the text hidden by these ellipsis).
-       * It could be also affected by the font itself, when it has wider render bouncing borders.
-       * So to be sure, that text container in slightly wider than parent,
-       * we have to to add some super-minor space at the end.
-       * Working in pair with a `fontRenderSpace` at useEllipsis hook.
-       */
-      paddingRight: '0.9px'
```
Comment (OleksandrNechai) [inline packages/picasso-lab/src/Ellipsis/Ellipsis.tsx:36]: Most likely this removal causes visual tests failure

File: packages/picasso-lab/src/TypographyOverflow/TypographyOverflow.tsx (hunk 1 above)
Comment (denieler) [reply packages/picasso-lab/src/TypographyOverflow/TypographyOverflow.tsx]: I know, but at the first place it was a code, which needed improvement

File: packages/picasso-lab/src/TypographyOverflow/test.tsx [hunk 3]
```diff
@@ -1,33 +1,56 @@
...
 
-jest.mock('@toptal/picasso-lab/Ellipsis/use-ellipsis', () => {
-  return jest.fn(() => ({
-    ref: null,
-    isEllipsis: true
-  }))
+jest.mock('@toptal/picasso-shared', () => {
+  return {
+    __esModule: true,
+    ...(jest.requireActual('@toptal/picasso-shared') as {}),
```
Comment (denieler) [inline packages/picasso-lab/src/TypographyOverflow/test.tsx:9]: which error it is reporting, when we don't have this casting to `as {}`?

File: packages/picasso-lab/src/TypographyOverflow/TypographyOverflow.tsx (hunk 1 above)
Comment (OleksandrNechai) [reply packages/picasso-lab/src/TypographyOverflow/TypographyOverflow.tsx]: Agree, although not sure if it is possible with shitty we tech :-)

File: packages/picasso-lab/src/TypographyOverflow/test.tsx (hunk 3 above)
Comment (OleksandrNechai) [reply packages/picasso-lab/src/TypographyOverflow/test.tsx:9]: Spread types may only be created from object types.ts(2698)

**Comment by alexanderpolyankin** [comment]
Fixed in the follow up PR https://github.com/toptal/picasso/pull/2061, closing this one.

**Pull request closed without merging**